src/App.tsx

```diff
@@ -1,93 +1,262 @@
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, {
+  useState,
+  useCallback,
+  useEffect,
+  useMemo,
+} from 'react';
+import {
+  deleteTodos,
+  getActiveTodos, getCompletedTodos, getTodos, patchTodos, postTodos,
+} from './api/todos';
+import { Todo } from './types/Todo';
+import { Error } from './components/Error/Error';
+import { TodoList } from './components/TodoList';
+import { FilterParams } from './Filter/FilterParams';
+import { Filter } from './Filter';
+import { TodoContext } from './components/TodoContext';
+
+const USER_ID = 9925;
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [inputValue, setInputValue] = useState('');
+  const [isGetDataError, setIsGetDataError] = useState(false);
+  const [isInputEmpty, setIsInputEmpty] = useState(false);
+  const [isPostError, setIsPostError] = useState(false);
+  const [isInputLocked, setIsLockInput] = useState(false);
+  const [isDeleteError, setIsDeleteError] = useState(false);
+  const [tempTodo, setTempTodo] = useState<string | null>(null);
+  const [filterParam, setFilterParam] = useState(
+    localStorage.getItem('filter') || FilterParams.All,
```

Comment:
Use `useEffect` to manage side effects like fetching from localStorage, so the component's body is kept pure.

Reply:
Fixed